File: docs-old-1103-2/extensibility/codesnippet/CSharp/walkthrough-implementing-code-snippets_10.cs
            //code previously written for Exec
            if (pguidCmdGroup == VSConstants.VSStd2K && nCmdID == (uint)VSConstants.VSStd2KCmdID.TYPECHAR)
            {
                typedChar = (char)(ushort)Marshal.GetObjectForNativeVariant(pvaIn);
            }
            //the snippet picker code starts here
            if (nCmdID == (uint)VSConstants.VSStd2KCmdID.INSERTSNIPPET)
            {
                IVsTextManager2 textManager = (IVsTextManager2)m_provider.ServiceProvider.GetService(typeof(SVsTextManager));

                textManager.GetExpansionManager(out m_exManager);

                m_exManager.InvokeInsertionUI(
                    m_vsTextView,
                    this,      //the expansion client
                    new Guid(SnippetUtilities.LanguageServiceGuidStr),
                    null,       //use all snippet types
                    0,          //number of types (0 for all)
                    0,          //ignored if iCountTypes == 0
                    null,       //use all snippet kinds
                    0,          //use all snippet kinds
                    0,          //ignored if iCountTypes == 0
                    "TestSnippets", //the text to show in the prompt
                    string.Empty);  //only the ENTER key causes insert 

                return VSConstants.S_OK;
            }

File: docs-old-1104/extensibility/codesnippet/VisualBasic/walkthrough-implementing-code-snippets_10.vb
            'code previously written for Exec
            If pguidCmdGroup = VSConstants.VSStd2K AndAlso nCmdID = CUInt(VSConstants.VSStd2KCmdID.TYPECHAR) Then
                typedChar = ChrW(CUShort(Marshal.GetObjectForNativeVariant(pvaIn)))
            End If
            'the snippet picker code starts here
            If nCmdID = CUInt(VSConstants.VSStd2KCmdID.INSERTSNIPPET) Then
                Dim textManager As IVsTextManager2 = DirectCast(m_provider.ServiceProvider.GetService(GetType(SVsTextManager)), IVsTextManager2)

                textManager.GetExpansionManager(m_exManager)
                m_exManager.InvokeInsertionUI(
                    m_vsTextView,
                    Me,
                    New Guid(SnippetUtilities.LanguageServiceGuidStr),
                    Nothing,
                    0,
                    0,
                    Nothing,
                    0,
                    0,
                    "TestSnippets",
                    String.Empty)
                Return VSConstants.S_OK
            End If